### DISCUSSION TITLE
I want to know how to distinguish between user editing operations and script implemented editing operations

### DISCUSSION THREAD

**Discussion by wangwangwang23333** [microsoft/vscode-discussions#1158] [Extension Development QnA] [answered]
In my plugin, I will retrieve user editing operations and send them to the server. At the same time, I will also receive the operations processed by the server and synchronize them to the local editor. I would like to know if there is any good way to distinguish between user manual editing operations and script executed editing operations.

**Comment by alefragnani** [comment, marked as answer]
Hi @wangwangwang23333 ,

You may be facing a similar situation as #1157, and if that's the case, if you have control/ownership of the server that processes the operations, I would suggest you to incorporate some kind of identification yourself on those editing operations, in order to identify when it comes from the server. 

Neither the `onDidChangeTextDocument` event, nor `TextEdit.edit` identifies who did execute it so, you can't depend on it to distinguish the source.

Hope this helps

**Comment by abrilurena** [comment]
Hi, @wangwangwang23333! do you still need help with this?

**Reply by wangwangwang23333** [reply]
So now there's no way to undentify this...☹️ thanks

**Reply by abrilurena** [reply]
Hi! i'm adding @isidorn - maybe he can help :)

**Reply by isidorn** [reply]
Nothing to add. I think @alefragnani answer is good.
